fleetshard/pkg/centralreconciler/reconciler_test.go

```diff
@@ -71,3 +73,111 @@ func TestReconcileUpdateSucceeds(t *testing.T) {
 	assert.Equal(t, centralName, central.GetName())
 	assert.Equal(t, "4", central.GetAnnotations()[revisionAnnotationKey])
 }
+
+func TestReconcileLastHashEmptyOnError(t *testing.T) {
+	fakeClient := testutils.NewFakeClientBuilder(t, &v1alpha1.Central{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      centralName,
+			Namespace: centralName,
+			// most straighforward way to simulat Reconcile returning an error
+			// is setting this value to something that is not a number
+			Annotations: map[string]string{revisionAnnotationKey: "NaN"},
```

Comment:
You can extract "NaN" to a constant named "invalidRevision" or add an invalid text directly to it to avoid the comment.
For example:

```suggestion
			Annotations: map[string]string{revisionAnnotationKey: "invalid annotation"},
```

Reply:
Done.